### PR TITLE
Move zwave_js node comments from device config to info page

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
This PR moves node comments from the device config page to device info page (see screenshot below). Aside from the fact that this now supports controller nodes, it is a better UX decision overall.

This PR may need some mobile friendly CSS help, I am not an expert. It also must be merged alongside the corresponding core PR (https://github.com/home-assistant/core/pull/71513)

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- Link to core pull request: https://github.com/home-assistant/core/pull/71513

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
Backend PR is approved.

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (zsarnett) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts:40]:
```suggestion
  protected willUpdate(changedProperties: PropertyValues) {
  	super.willUpdate(changedProps);
```

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (zsarnett) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts:54]:
```suggestion
  private async _fetchNodeDetails() {
```
Also should be moved below render

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (zsarnett) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts:84]:
```suggestion
    if (this._nodeComments?.comments?.length <= 0) {
```

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (bramkragten) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts:51]: Check https://github.com/home-assistant/frontend/pull/12642

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (zsarnett) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts:32]: Do we need to store this? Where is it used? Seems only in the for loop?

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (raman325) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts]: ts gives an error that this may be undefined

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (raman325) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts:98]: doesn't appear to be needed  👍🏾

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts
Comment (raman325) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts]: seems to work though so maybe that's ok?

Comment (raman325) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-alerts-zwave_js.ts]: I reverted this and the type checker is happy. Let me know if there's another better way to resolve that issue

**Comment by bramkragten** [comment]
Should we move these warning to the left column with device info?

**Comment by raman325** [comment]
I'm putting this on hold until the other PR to use device ID in the front end is approved (backend is approved) so that I can refactor. Once that's done I will try the left side to see how that looks. My only concern is if the text is too long, it may not look great

**Comment by raman325** [comment]
here are the two implementations side by side:
image

I replaced the test text with the longest comment I could find. Based on this I think it's better to keep it across the top, not just in the card.

In the card: 
image

At the top: 
image

**Comment by matthiasdebaat** [comment]
> I replaced the test text with the longest comment I could find (this is something that would actually be shown to users once this feature gets released). Based on this I think it's better to keep it across the top, not just in the card.
> 
> In the card: image
> 
> At the top: image

In general, I think it's better to have it in the card, because it's the most relevant. 

This example is indeed quite a long piece of text. I would recommend to change the text and add a button. For example:
`There is a known bug prior to version 7.17.2 that caused the network and controller to become unresponsive. [How to upgrade?]`

**Comment by zsarnett** [comment]
Can you place this back into PR state once the backend is merged?

**Comment by MartinHjelmare** [comment]
We normally wait with merging the backend PR until the frontend PR is approved. The backend PR is approved already. We want to make sure that the frontend has everything it needs before we merge the backend PR.

**Comment by zsarnett** [comment]
Sorry didnt look to see it was approved. My bad.

**Comment by raman325** [comment]
> This example is indeed quite a long piece of text. I would recommend to change the text and add a button. For example:
> 
> `There is a known bug prior to version 7.17.2 that caused the network and controller to become unresponsive. [How to upgrade?]`

We don't have control over the text, it's provided by zwave-js

**Comment by zsarnett** [comment]
I am not sure I like the in the card design. Would rather have it on top of that column in stead... With the alert need a lot of width and currently its hitting the edges of the card which doesn't look great...

Can we put it in the top of the first column maybe that is a mix of both?

**Comment by raman325** [comment]
here's what it looks like above the device info card in the first column:
image

I personally still like the alert across the top as I had originally written it, but obviously I have some bias given that's what I started with :)

File: src/panels/config/devices/ha-config-device-page.ts
Comment (bramkragten) [inline src/panels/config/devices/ha-config-device-page.ts:552]: There is no `alerts` slot is there?
```suggestion
                  <div class="fullwidth">
```

**Review by bramkragten — DISMISSED** [review]

**Comment by raman325** [comment]
final screenshot with everything cleaned up: 
image

**Review by zsarnett — APPROVED** [review]

**Comment by zsarnett** [comment]
Good to Go whenever the backend is merged 👍

**Comment by raman325** [comment]
core PR merged

**Pull request merged**
